lib/dry/schema/result.rb

```diff
@@ -12,10 +12,38 @@ class Result
       alias_method :to_h, :output
       alias_method :to_hash, :output
 
-      param :results
+      param :results, default: -> { EMPTY_ARRAY.dup }
 
       option :message_compiler
 
+      def self.new(*args)
+        result = super
+        yield(result)
+        result.freeze
+      end
+
+      def set(hash)
```

Comment:
I wonder if `replace` is a more intention-revealing name here?

I think it would certainly help me understand the usage inside `Processor#call` a little better, e.g.

```ruby
result.replace(output) if output.is_a?(::Hash)
```

This would give us a nice naming alignment with the Ruby core language, e.g. `Array#replace` and `Hash#replace`.